tests/e2e/test_e2e.py

```diff
@@ -297,6 +408,11 @@ def forward(self, input):
         for idx in range(len(test2_pred["outputs"]["task2"]["_input__data"]))
     ]
 
+    assert len(test4_pred["outputs"]["task3"]["input1_t3_out"]["image_pil"]) == 10
+    assert isinstance(
+        test4_pred["outputs"]["task3"]["input1_t3_out"]["image_pil"], np.ndarray
```

Comment:
Can we check the shape?

Reply:
for https://github.com/senwu/emmental/pull/113#discussion_r740705320, we can't get rid of it because `out_dict[task_name][action_name]` and `out_bdict[task_name][action_name]` may not be of the same type for the empty check to be handled by `merge_objects`